### PR TITLE
Move TX History to Redux

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

File: src/services/Store/StoreProvider.tsx [hunk 1]
```diff
@@ -240,11 +236,7 @@ export const StoreProvider: React.FC = ({ children }) => {
 
   useEffect(() => {
     if (mainnetAccounts.length > 0) {
-      HistoryService.instance.getHistory(mainnetAccounts).then((history) => {
-        if (history !== null) {
-          setTxHistory(history);
-        }
-      });
+      dispatch(fetchHistory());
```
Comment (GuillaumeRx) [inline src/services/Store/StoreProvider.tsx:239]: Can we move this conditional in the saga and dispatch the event at `handleRehydrateSuccess` in `persistence.slice.ts` ?

**Review by GuillaumeRx — CHANGES_REQUESTED** [review]

**Comment by FrederikBolding** [comment]
[ch8342]

File: src/services/Store/StoreProvider.tsx (hunk 1 above)
Comment (FrederikBolding) [reply src/services/Store/StoreProvider.tsx:239]: We need it to run every time the accounts change though. So that is not enough.

Comment (FrederikBolding) [reply src/services/Store/StoreProvider.tsx:239]: Possibly. Will look into this.

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**